src/overlays/actors/ovl_En_Niw_Girl/z_en_niw_girl.c

```diff
@@ -21,21 +31,226 @@ const ActorInit En_Niw_Girl_InitVars = {
     (ActorFunc)EnNiwGirl_Update,
     (ActorFunc)EnNiwGirl_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw_Girl/EnNiwGirl_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Niw_Girl/EnNiwGirl_Destroy.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x00, 0x39, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 10, 30, 0, { 0, 0, 0 } },
+};
+
+static Vec3f sConstVec3f = { 0.2f, 0.2f, 0.2f };
+
+Gfx* D_80AB99D8[] = { 0x06004178, 0x06004978, 0x06005178, 0x00000000, 0x00000000, 0x00000000 };
+
+extern SkeletonHeader D_06009948;
+extern AnimationHeader D_06000378;
+extern AnimationHeader D_06009C78;
+
+void EnNiwGirl_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnNiwGirl* this = THIS;
+    s32 pad;
+    Vec3f tempSrc;
+    Vec3f tempDst;
+    s32 pad2;
+
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06009948, &D_06000378, &this->limbDrawTable,
+                     &this->transitionDrawTable, 17);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, thisx, &sCylinderInit);
+    thisx->unk_1F = 6;
+    if (thisx->params < 0) {
+        thisx->params = 0;
+    }
+    this->unkFlag = ((thisx->params >> 8) & 0xFF);
+    thisx->gravity = -3.0f;
+    Matrix_RotateY((thisx->shape.rot.y / 32768.0f) * M_PI, 0);
+    tempDst.x = tempDst.y = tempDst.z = 0.0f;
+    tempSrc.x = tempSrc.y = 0.0f;
+    tempSrc.z = 50.0;
+    Matrix_MultVec3f(&tempSrc, &tempDst);
+    this->attachedActor = (EnNiw*)Actor_SpawnAttached(&globalCtx->actorCtx, thisx, globalCtx, ACTOR_EN_NIW,
+                                                      thisx->posRot.pos.x + tempDst.x, thisx->posRot.pos.y + tempDst.y,
+                                                      thisx->posRot.pos.z + tempDst.z, 0, thisx->posRot.rot.y, 0, 0xA);
+    if (this->attachedActor != 0) {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ シツレイしちゃうわね！プンプン ☆☆☆☆☆ %d\n" VT_RST, thisx->params);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ きゃははははは、まてー ☆☆☆☆☆ %d\n" VT_RST, this->unkFlag);
+        osSyncPrintf("\n\n");
+        thisx->colChkInfo.mass = 0xFF;
+        this->actionFunc = EnNiwGirl_Talk;
+    } else {
+        osSyncPrintf("\n\n");
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ なぜか、セットできむぅあせん ☆☆☆☆☆ %d\n" VT_RST, thisx->params);
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ んんがくく ☆☆☆☆☆ %d\n" VT_RST, this->unkFlag);
+        osSyncPrintf("\n\n");
+        Actor_Kill(thisx);
+    }
+}
+
+void EnNiwGirl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnNiwGirl_PlayJumpAnimation(EnNiwGirl* this, GlobalContext* globalCtx) {
```

Comment:
The name for this action should just be something like "Jump" imo (playing the animation being part of the action's implementation)
```suggestion
void EnNiwGirl_Jump(EnNiwGirl* this, GlobalContext* globalCtx) {
```